src/utils/FixedQueue.ts

```diff
@@ -0,0 +1,55 @@
+import * as _ from 'underscore';
```

Comment:
thanks for the article, interesting indeed. 

Trying to be a bit less academic and pragmatic here, the problem I see is that *you don't use your method `contains`  and `removeElement` anywhere at all*,  I suspect because you realized they don't work well with objects and arrays. So two choices here: 1) you reimplement the methods so they work with complex data structures, 2) you remove them along with their unit tests since you are not using them and sleep at night because we have everything we need to perform these actions in our stack anyways. 

For your `getFirstElement`, this is like saying you would need a method `getQueueLength()` to do `this.getQueue().length`...do we really want to abstract one liners like this all the time?

Here are other principles to reflect on:
1. Keep it simple stupid
2. Less code is better/Best code is no code
3. don't repeat yourself
4. https://softwareengineering.stackexchange.com/questions/202477/can-too-much-abstraction-be-bad